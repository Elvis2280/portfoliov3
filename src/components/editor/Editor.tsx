import React, { ReactElement } from 'react'
import dynamic from 'next/dynamic'
import 'react-quill/dist/quill.snow.css'
import { Box } from '@chakra-ui/react'

const ReactQuill = dynamic(() => import('react-quill'), { ssr: false })

type EditorProps = {
  value?: string
  handlerText: (value: string) => void
}

const Editor = ({ handlerText, value }: EditorProps): ReactElement => {
  const handleChange = (value: string): void => {
    handlerText(value)
  }

  return (
    <Box display={'block'} position={'relative'} height={'400px'}>
      <ReactQuill
        value={value}
        onChange={handleChange}
        theme='snow'
        style={{ height: '400px' }}
        modules={{
          toolbar: [
            [{ header: [1, 2, false] }],
            ['bold', 'italic', 'underline', 'strike', 'blockquote'],
            [
              { list: 'ordered' },
              { list: 'bullet' },
              { indent: '-1' },
              { indent: '+1' },
            ],
            ['link', 'image'],
            ['clean'],
          ],
        }}
      />
    </Box>
  )
}

export default Editor
