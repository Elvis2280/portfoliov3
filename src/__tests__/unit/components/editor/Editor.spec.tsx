import React from 'react'
import { render } from '@testing-library/react'
import Editor from '@/components/editor/Editor'

describe('Editor', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Editor handlerText={() => {}} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
