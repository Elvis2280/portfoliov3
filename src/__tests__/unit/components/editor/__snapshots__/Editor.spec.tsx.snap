// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-urh063"
  />
</DocumentFragment>
`;
